include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -961,135 +970,449 @@ void reduceSumScalingVector(common::Pipeline<comm::Communicator>& col_task_chain
+      const comm::IndexT_MPI evecs_tile_rank = dist.rankGlobalTile<Coord::Col>(i);
+      auto& tile = eval_tiles[to_sizet(i - i_begin)];
+
+      if (evecs_tile_rank == this_rank.col())
+        comms.emplace_back(ex::when_all(row_comm_chain(), ex::just(std::cref(tile))) |
+                           transformMPI(sendBcast_o));
+      else
+        comms.emplace_back(ex::when_all(row_comm_chain(), ex::just(evecs_tile_rank, std::cref(tile))) |
+                           transformMPI(recvBcast_o));
+    }
+
+    return ex::ensure_started(ex::when_all_vector(std::move(comms)));
+  };
+
+  auto all_reduce_in_place_o = [](const dlaf::comm::Communicator& comm, MPI_Op reduce_op,
+                                  const auto& data, MPI_Request* req) {
+    auto msg = comm::make_message(data);
+    DLAF_MPI_CHECK_ERROR(MPI_Iallreduce(MPI_IN_PLACE, msg.data(), msg.count(), msg.mpi_type(), reduce_op,
+                                        comm, req));
+  };
+
+  // Note: at least two column of tiles per-worker, in the range [1, getTridiagRank1NWorkers()]
+  const std::size_t nthreads = [size = sz_loc_tiles.cols()]() {
+    const std::size_t min_workers = 1;
+    const std::size_t available_workers = getTridiagRank1NWorkers();
+    const std::size_t ideal_workers = util::ceilDiv(to_sizet(size), to_sizet(2));
+    return std::clamp(ideal_workers, min_workers, available_workers);
+  }();
+
+  ex::start_detached(
+      ex::when_all(ex::just(std::make_unique<pika::barrier<>>(nthreads)),
+                   std::forward<CommSender>(row_comm), std::forward<CommSender>(col_comm),
+                   std::forward<KSender>(k), std::forward<RhoSender>(rho),
+                   ex::when_all_vector(tc.read(d)), ex::when_all_vector(tc.readwrite(z)),
+                   ex::when_all_vector(tc.readwrite(evals)), ex::when_all_vector(tc.read(i2)),
+                   ex::when_all_vector(tc.readwrite(evecs)),
+                   // additional workspaces
+                   ex::just(std::vector<memory::MemoryView<T, Device::CPU>>()),
+                   ex::just(memory::MemoryView<T, Device::CPU>())) |
+      ex::transfer(di::getBackendScheduler<Backend::MC>(pika::execution::thread_priority::high)) |
+      ex::bulk(nthreads, [nthreads, n, n_subm_el_lc, m_subm_el_lc, i_begin, ij_begin_lc, sz_loc_tiles,
+                          dist, bcast_evals, all_reduce_in_place_o](
+                             const std::size_t thread_idx, auto& barrier_ptr, auto& row_comm_wrapper,
+                             auto& col_comm_wrapper, const auto& k, const auto& rho,
+                             const auto& d_tiles_futs, auto& z_tiles, const auto& eval_tiles,
+                             const auto& i2_tile_arr, const auto& evec_tiles, auto& ws_cols,
+                             auto& ws_row) {
+        using dlaf::comm::internal::transformMPI;
+
+        common::Pipeline<comm::Communicator> row_comm_chain(row_comm_wrapper.get());
```

Comment:
AFAIR this was not strictly needed, because we are launching them in order in a deterministic way...at least from our point of view. There might be details that I'm not  aware of: e.g. I submit tasks in order to a thread queue, but than it execute them in a different order).

In the end it looked safer and cleaner to use the pipeline.

Reply:
I think this was indeed a potential issue
> I submit tasks in order to a thread queue, but than it execute them in a different order
so I think this is a good change. Thanks for explaining!